.github/workflows/bench.yml

```diff
@@ -0,0 +1,60 @@
+name: benchmarks
+
+on: push
+
+jobs:
+  bench:
+    name: Benchmarks
+    runs-on: ubuntu-latest
+
+    container:
+      image: ghcr.io/n3pdf/bench-evol:latest
+      credentials:
+        username: ${{ github.repository_owner }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          # tags needed for dynamic versioning
+          fetch-depth: 0
+      - name: Install and configure Poetry
+        uses: snok/install-poetry@v1
+        with:
+          virtualenvs-path: ~/.virtualenvs
+          installer-parallel: true
+      - name: Cache Poetry virtualenv
+        uses: actions/cache@v2
+        id: cache
+        with:
+          path: ~/.virtualenvs
+          key: ${{ runner.os }}-py-${{ steps.setup-python.outputs.python-version }}-poetry-${{ hashFiles('poetry.lock') }}
+      - name: Install version management tool
+        run: |
+          # same poetry env
+          PIP="$(head -n1 $(which poetry) | cut -c 3-) -m pip"
+          ${PIP} install poetry-dynamic-versioning
+      - name: Install dependencies
+        run: poetry install --no-interaction --no-root
+        if: steps.cache.outputs.cache-hit != 'true'
+      - name: Install project
+        run: |
+          poetry install --no-interaction ${{ inputs.poetry-extras }}
```

Comment:
```suggestion
      - name: Install dependencies
        run: poetry install --no-interaction --no-root ${{ inputs.poetry-extras }}
        if: steps.cache.outputs.cache-hit != 'true'
      - name: Install project
        run: |
          poetry install --no-interaction --only-root
```

Reply:
Explanation: the extras should go together with the other dependencies.

Moreover, if everything is already fine and skipped, the extra step should only install the project.